frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/dialogs/GitPushDialog.java

```diff
@@ -58,128 +25,22 @@ public GitPushDialog(
       List<? extends Repository> selectedRepositories,
       PushSource pushSource,
       boolean isForcePushRequired) {
-    super(project, /* canBeParent */ true,
-        Registry.is("ide.perProjectModality") ? IdeModalityType.PROJECT : IdeModalityType.IDE);
+    // Presented dialog shows commits for branches belonging to allRepositories, preselectedRepositories and currentRepo.
+    // The second and the third one have a higher priority of loading its commits.
+    // From our perspective, we always have a single (pre-selected) repository, so we do not care about the priority.
+    super(project, selectedRepositories.asJava(), selectedRepositories.asJava(), /* currentRepo */ null, pushSource);
     this.isForcePushRequired = isForcePushRequired;
-    this.project = project;
-
-    MyVcsPushDialog dialog = new MyVcsPushDialog(project, selectedRepositories.asJava(), pushSource);
-    this.pushController = dialog.getPushController();
-    this.listPanel = pushController.getPushPanelLog();
-
     this.pushAction = new PushSwingAction();
-    this.pushAction.putValue(DEFAULT_ACTION, Boolean.TRUE);
 
     // Note: since the class is final, `this` is already @Initialized at this point.
 
-    dialog.updateOkActions();
     setOKButtonText(getPushActionName());
-    setTitle("Push Commits");
     init();
   }
 
-  private class MyVcsPushDialog extends VcsPushDialog {
```

Comment:
Also... can `com.virtuslab.archunit.ClassStructureTestSuite#inner_classes_should_not_be_instantiated_from_constructor_of_enclosing_class` be updated now?

Reply:
sadly no 😢 there is still this line:
```java
this.pushAction = new PushSwingAction()
```